pkg/collector/osm_collector.go

```diff
@@ -0,0 +1,189 @@
+package collector
+
+import (
+	"log"
+	"path/filepath"
+	"regexp"
+
+	"github.com/Azure/aks-periscope/pkg/interfaces"
+	"github.com/Azure/aks-periscope/pkg/utils"
+)
+
+// OsmCollector defines an Osm Collector struct
+type OsmCollector struct {
+	BaseCollector
+}
+
+var _ interfaces.Collector = &OsmCollector{}
+
+// NewOsmCollector is a constructor
+func NewOsmCollector(exporter interfaces.Exporter) *OsmCollector {
+	return &OsmCollector{
+		BaseCollector: BaseCollector{
+			collectorType: Osm,
+			exporter:      exporter,
+		},
+	}
+}
+
+// Collect implements the interface method
+func (collector *OsmCollector) Collect() error {
+	// Get all OSM deployments in order to collect information for various resources across all meshes in the cluster
+	meshList, err := utils.GetResourceList([]string{"get", "deployments", "--all-namespaces", "-l", "app=osm-controller", "-o", "jsonpath={..meshName}"}, " ")
+	if err != nil {
+		return err
+	}
+
+	// Directory where OSM logs will be written to
+	rootPath, err := utils.CreateCollectorDir(collector.GetName())
+	if err != nil {
+		return err
+	}
+
+	for _, meshName := range meshList {
```

Comment:
💡 What is this Range trying to  achieve in the OSM collector context here? (Reason I ask is because if I correctly understand the needs, we could simplify this)

Reply:
This is needed in case there are multiple instances of OSM installed on the cluster. In that case we would want to collect everything for each of the OSMs